auth/authentication.md

```diff
@@ -0,0 +1,61 @@
+# Authentication
+
+## User Flow
+
+User visits a new website that requires authentication from wallet.
+
+1. Website displays qrcode or deep link for auth
+2. User scans qrcode or redirects to wallet
+3. User approves prompt for auth in wallet
+4. User returns to website after prompt success
+5. Website is now authenticated with signed message
+
+User returns to a previously visted website before authentication expiry
+
+1. Website checks if user is authenticated from local storage
+2. Website is now authenticated with signed message
+
+User returns to a previously visted website after authentication expiry
+
+1. Website checks if user is authenticated from local storage
+2. Website sends new request to previously paired wallet
+3. User views notification and redirects to wallet
+4. User approves prompt for auth in wallet
+5. User returns to website after prompt success
+6. Website is now authenticated with signed message
+
+## Request Protocol
+
+A creates random symKey S for pairing topic.
+
+Pairing topic is the hash of symKey S.
+
+A generates keyPair X and generates response topic.
+
+Response topic is the hash of publicKey X.
+
+A will construct an authentication request.
+
+A encrypts request with symKey S.
+
+A publishes encrypted request to topic.
+
+A starts polling messages on response topic.
```

Comment:
polling? - so we are planning to not have a websocket on A?

Reply:
By polling you mean a GET HTTP where topic is added as query?
Or POST HTTP with JsonRpcRequest as body where topic is inside params?

Reply:
Correct... we need to coordinate this with Iridium team to include a `irn_getMessages` endpoint